src/utils/index.ts

```diff
@@ -49,6 +50,32 @@ export interface Route {
   interceptor?: (req: Request, res: Response, next: NextFunction) => void;
 }
 
+async function pgTx<T>(pool: Pool, cb: (client: ClientBase) => Promise<T>): Promise<T> {
+  const client = await pool.connect();
+  try {
+    await client.query("BEGIN");
+    const result = await cb(client);
+    await client.query("COMMIT");
```

Comment:
@vsubhuman , if the idea here is just to take the snapshot, I'd not commit, but rollback, I'm not familiar with `pg` isolation levels, but in some scenarios if you try to commit and the records in the snapshot have changed it might fail, which I think is not necessary.

Reply:
good point, updating